kore/src/Kore/Builtin/Bool.hs

```diff
@@ -163,48 +165,63 @@ builtinFunctions =
     xor a b = (a && not b) || (not a && b)
     implies a b = not a || b
 
+data UnifyBool = UnifyBool
+    { bool1, bool2 :: InternalBool
+    }
+
+matchBools ::
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    Maybe UnifyBool
+matchBools first second
+    | InternalBool_ bool1 <- first
+      , InternalBool_ bool2 <- second =
+        Just $ UnifyBool bool1 bool2
+    | otherwise = Nothing
+{-# INLINE matchBools #-}
+
 -- | Unification of @BOOL.Bool@ values.
 unifyBool ::
-    forall variable unifier.
-    InternalVariable variable =>
+    forall unifier.
     MonadUnify unifier =>
-    TermLike variable ->
-    TermLike variable ->
-    MaybeT unifier (Pattern variable)
-unifyBool a b =
-    worker a b <|> worker b a
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    UnifyBool ->
+    unifier (Pattern RewritingVariableName)
+unifyBool termLike1 termLike2 unifyData =
+    if bool1 == bool2
+        then return (Pattern.fromTermLike termLike1)
+        else
+            Unify.explainAndReturnBottom
+                "different Bool domain values"
+                termLike1
+                termLike2
```

Comment:
```suggestion
unifyBool termLike1 termLike2 unifyData
  | bool1 == bool2 = return (Pattern.fromTermLike termLike1)
  | otherwise =
    Unify.explainAndReturnBottom
        "different Bool domain values"
        termLike1
        termLike2
```